src/System Application/App/Security Groups/src/SecurityGroupImpl.Codeunit.al

```diff
@@ -362,6 +364,31 @@ codeunit 9871 "Security Group Impl."
         end;
     end;
 
+    procedure GetCode(GroupId: Text[250]; var GroupCode: Code[20]): Boolean
+    var
+        User: Record User;
+        UserProperty: Record "User Property";
+        SecurityGroup: Record "Security Group";
+    begin
+        if IsWindowsAuthentication() then begin
+            User.SetRange("Windows Security ID", GroupId);
+            if not User.FindFirst() then
+                exit(false);
+            SecurityGroup.SetRange("Group User SID", User."User Security ID");
+        end else begin
+            UserProperty.SetRange("Authentication Object ID", GroupId);
```

Comment:
Why does it check against the UserProperty?

Reply:
It's the related table where the ID is actually stored.